packages/tlinks/src/ext/twitter.tsx

```diff
@@ -1,372 +1,432 @@
-    }),
-  );
+	addMargin(container).replaceChildren(
+		actionElement
+	);
 }
 
 function createTokenScriptIframe(
-  websiteUrl: string,
-  renderer: ActionAdapter['tsIframeRenderer'],
+	websiteUrl: string,
+	renderer: ActionAdapter['tsIframeRenderer'],
 ) {
-  const container = document.createElement('div');
-  container.className = 'dialect-action-root-container';
+	const container = document.createElement('div');
+	container.className = 'dialect-action-root-container';
 
-  const actionRoot = createRoot(container);
-  const Comp = renderer || (() => null);
+	const actionRoot = createRoot(container);
+	const Comp = renderer || (() => null);
 
-  actionRoot.render(
-    <div onClick={(e) => e.stopPropagation()}>
-      <Comp websiteUrl={websiteUrl} />
-    </div>,
-  );
+	actionRoot.render(
+		<div onClick={(e) => e.stopPropagation()}>
+			<Comp websiteUrl={websiteUrl} />
+		</div>,
+	);
 
-  return container;
+	return container;
 }
 
 function createAction({
-  originalUrl,
-  action,
-  callbacks,
-  options,
+	originalUrl,
+	action,
+	callbacks,
+	options,
 }: {
-  originalUrl: URL;
-  action: Action;
-  callbacks: Partial<ActionCallbacksConfig>;
-  options: NormalizedObserverOptions;
-  isInterstitial: boolean;
+	originalUrl: URL;
+	action: Action;
+	callbacks: Partial<ActionCallbacksConfig>;
+	options: NormalizedObserverOptions;
+	isInterstitial: boolean;
 }) {
-  const container = document.createElement('div');
-  container.className = 'dialect-action-root-container';
-
-  const actionRoot = createRoot(container);
-
-  actionRoot.render(
-    <div onClick={(e) => e.stopPropagation()}>
-      <ActionContainer
-        stylePreset={resolveXStylePreset()}
-        action={action}
-        websiteUrl={originalUrl.toString()}
-        websiteText={originalUrl.hostname}
-        callbacks={callbacks}
-        securityLevel={options.securityLevel}
-      />
-    </div>,
-  );
-
-  return container;
+	const container = document.createElement('div');
+	container.className = 'dialect-action-root-container';
+
+	const actionRoot = createRoot(container);
+
+	actionRoot.render(
+		<div onClick={(e) => e.stopPropagation()}>
+            <ActionContainer
+                stylePreset={resolveXStylePreset()}
+                action={action}
+                websiteUrl={originalUrl.toString()}
+                websiteText={originalUrl.hostname}
+                callbacks={callbacks}
+                securityLevel={options.securityLevel}
+            />		
+		</div>,
+	);
+
+	return container;
+}
+
+function createFarcasterFrame({
+	chain,
+	scriptURI,
+}: {
+	chain: number;
+	scriptURI: string;
+}) {
+
+	const container = document.createElement('div');
+	container.className = 'dialect-action-root-container';
+	const actionRoot = createRoot(container);
+	const queryClient = new QueryClient();
+
+	actionRoot.render(
+		<div onClick={(e) => e.stopPropagation()}>
+            <WagmiProvider config={wagmiProviderConfig}>
```

Comment:
the user is not able to connect to wallet in this "tlink" package, or in the context of "ISOLATED" world of chrome extension.
please remove any wallet related code, including the rainbowkit deps from the package.json